src/test/ui/impl-trait/hidden-lifetimes.rs

```diff
@@ -0,0 +1,53 @@
+// Test to show actual unsound behaviour that occurs when the hidden lifetime
+// check is removed from impl trait types.
```

Comment:
Can you cc the relevant issues/PRs here in the test, to help us track down more info later?

Reply:
Or else expand this comment a bit?

Reply:
Or both? :)

Reply:
Something like:

---

Test to show what would happen if we were not careful and permitted lifetimes to escape through an impl trait. In this case, we are returning a `&'a mut &'b T`, but the `'b` does not appear in the impl trait type. We can use this to engineer an unsoundness. (XXX perhaps a more detailed explanation)